### ISSUE TITLE
require('json-api') on io.js 3.x and 4.x leads to segmentation fault

### ISSUE THREAD

**Issue by capricube** [ethanresnick/json-api#76]
requiring the json-api on io.js greater than version 2 leads to segmentation fault: 11.

This simple program illustrates the problem:
```
const jsonApi = require('json-api');

console.log("loaded jsonApi");
```

However requiring the modules within the index.js works correctly. 

Any hints why this is failing?
Failure reproducible on Io.js 3.1, 3.2, 3.3 and 4.0-rc1

**Comment by capricube** [comment]
Update: after rewriting index.js i realized that the missing MongooseAdapter is causing the Problem. 

Is it enough just to remove the dbAdapters?

**Comment by ethanresnick** [comment]
Can you explain what you mean by "the missing MongooseAdapter"? The adapter shouldn't be missing...

**Comment by capricube** [comment]
we don't have the MongooseAdapter as dependency in our module because we don't use MongoDB. And json-api defines it as peer-dependency.

**Comment by ethanresnick** [comment]
Ah, gotcha. Then yes, you can just remove those exports from index.js. That's a bit hacky, though, so I'll think more tomorrow about the "right" solution, or would welcome a PR if you have any ideas

**Comment by capricube** [comment]
did you have any better idea to fix the problem? 

My PR will load the MongooseAdapter only if` `mongoose` is available.

**Comment by ethanresnick** [comment]
PR merged. Thanks!

**Comment by ethanresnick** [comment]
Published to npm as json-api@2.9.5

**Comment by capricube** [comment]
Thanks :+1:
